domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs

```diff
@@ -110,6 +117,18 @@ protected override IHost CreateHost(IHostBuilder builder)
         return client;
     }
 
+    public (WalletService.WalletServiceClient, Metadata metadata) CreateWalletClient(string subject)
+    {
+        var authentication = new AuthenticationHeaderValue("Bearer", GenerateToken(sub: subject));
+        var metadata = new Metadata { { "Authorization", $"{authentication.Scheme} {authentication.Parameter}" } };
+
+        var options = Services.GetRequiredService<IOptions<WalletOptions>>().Value;
```

Comment:
Could this be Dependency Injected? Or us the WalletUrl at line 42? I dont think it has to be so dynamic in test

Reply:
Corrected to use WalletUrl on line 42